tests/testthat/test-as.network.numeric.R

```diff
@@ -0,0 +1,81 @@
+  expect_silent(as.network.numeric(x=5, directed = FALSE))
+})
+test_that("Generating Bipartite network", {
+  expect_silent(as.network.numeric(x=5, bipartite = 3, directed = FALSE))
+})
+test_that("Stop when the loop == TRUE", {
+  expect_error(as.network.numeric(x=5, loops = TRUE, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, loops = TRUE, directed = TRUE))
+  expect_error(as.network.numeric(x=5, loops = TRUE, directed = FALSE))
+})
+test_that("Stop when the multiple == TRUE", {
+  expect_error(as.network.numeric(x=5, multiple = TRUE, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, multiple = TRUE, directed = TRUE))
+  expect_error(as.network.numeric(x=5, multiple = TRUE, directed = FALSE))
+})
+test_that("Stop when the hyper == TRUE", {
+  expect_error(as.network.numeric(x=5, hyper = TRUE, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, hyper = TRUE, directed = TRUE))
+  expect_error(as.network.numeric(x=5, hyper = TRUE, directed = FALSE))
+})
+test_that("Stop when the density is out of [0,1]", {
+  expect_error(as.network.numeric(x=5, density = -0.1, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, density = -0.1, directed = TRUE))
+  expect_error(as.network.numeric(x=5, density = -0.1, directed = FALSE))
+  expect_error(as.network.numeric(x=5, density =  1.1, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, density =  1.1, directed = TRUE))
+  expect_error(as.network.numeric(x=5, density =  1.1, directed = FALSE))
+})
+test_that("Stop when the number of edges is not integer", {
+  expect_error(as.network.numeric(x=5, numedges =  2.5, bipartite = NULL))
+  expect_error(as.network.numeric(x=5, numedges =  2.5, directed = TRUE))
+  expect_error(as.network.numeric(x=5, numedges =  2.5, directed = FALSE))
+})
+
+## # TODO: Add this back when explicit directed && bipartite is an error.
+##
+## test_that("Stop when a Directed bipartite network is specified by the user", {
+##   expect_error(as.network.numeric(x=5, numedges =  6L, directed = TRUE, bipartite = 3))
+## })
+
+#test_that("Stop when it is missed by the user to specify whether the network is bipartite or directed", {
+#  expect_error(as.network.numeric(x=5, numedges =  6L))
+#})
+
+## # TODO: Add this back when warnings can be relied on.
+## test_that("Warning message when bipartite == TRUE and user missed to specify whether the network is directed or undirected", {
+##   expect_warning(as.network.numeric(x=5, numedges =  6L, bipartite = 3))
+## })
+
+test_that("Correct number of edges for bipatite graph when numedges is passed", {
+  expect_silent(as.network.numeric(x=5, bipartite = 3, directed = FALSE))
+})
+test_that("Correct number of edges for undirected graph when numedges is passed", {
+  expect_silent(as.network.numeric(x=5, directed = FALSE))
+})
+test_that("Correct number of edges for directed graph when numedges is passed", {
+  expect_silent(as.network.numeric(x=5, directed = TRUE))
+})
```

Comment:
These don't check the number of edges.

Reply:
Thanks for the comment. Since the numedges are drawn from rbinom in which density is as an input, I think because of randomness in numedges it can't be checked unless for density =1 or density = 0.

The following are correct?

test_that("Correct number of edges for bipatite graph when numedges is passed", {
  expect_equal(network.edgecount(as.network.numeric(x=5, bipartite = 3, directed = FALSE, density = 1)),6)
  expect_equal(network.edgecount(as.network.numeric(x=5, bipartite = 3, directed = FALSE, density = 0)),0)
})
test_that("Correct number of edges for undirected graph when numedges is passed", {
  expect_equal(network.edgecount(as.network.numeric(x=5, directed = FALSE, density = 1)),10)
  expect_equal(network.edgecount(as.network.numeric(x=5, directed = FALSE, density = 0)),0)
})
test_that("Correct number of edges for directed graph when numedges is passed", {
  expect_equal(network.edgecount(as.network.numeric(x=5, directed = TRUE, density = 1)),20)
  expect_equal(network.edgecount(as.network.numeric(x=5, directed = TRUE, density = 0)),0)
})

Reply:
The function takes `numedges` as an argument, and that should fix the edges, right?

Reply:
Thanks for your comment. I'll correct the code base on scenario 1 as follows.

In fact, there are two scenarios: The user specified the number of edges or passed it.
1)	The numedges are specified: The number of edges in the generated network should be equal to the specified number.

test_that("Correct number of edges for bipatite graph",{
expect_equal(network.edgecount(as.network.numeric(x=5, bipartite = 3, directed = FALSE, numedges = 4)),4)
})
test_that("Correct number of edges for undirected graph",{
expect_equal(network.edgecount(as.network.numeric(x=5, directed = FALSE,  numedges = 10)),10)
})
test_that("Correct number of edges for directed graph",{
expect_equal(network.edgecount(as.network.numeric(x=5, directed = TRUE, numedges =  18)),18)
})

2)	The numedges are passed: The function uses density and the number of nodes to randomly generate the number of edges. I think your previous comment was about this scenario? (In this scenario for density =0 and density =1 we can check the number of edges in the generated network)

"Correct number of edges for directed, undirected, and bipartite networks if numedges is passed, for reasonable numedges."

Thanks for your time and guidance.

Reply:
I think cases in 1. already test for cases in 2.

Reply:
Yes, I will update based on scenario 1, it can cover scenario 2.